From 1671972d0f1bfeed4ac2243cefd90ca169f334a9 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Wed, 12 Jul 2023 10:32:54 +0800
Subject: [PATCH 1/2] puppet-manifests: fix the hardcoded kernel name

The kernel names in puppet-update-grub-env.py is hardcoded
as amd64, use dpkg-architecture to check host arch and
replace the hardcoded names.

Test Plan:
PASS: build-pkgs on x86-64 host
PASS: build-image on x86-64 host
PASS: build-pkgs on arm64 host
PASS: build-image on arm64 host
PASS: install AIO-SX on x86-64 targets
PASS: install AIO-SX on arm64 targets

Story: 2010739
Task: 48284

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
Change-Id: I1df220f585f53e58f2fcd8f6918273f8ea45fb90
---
 puppet-manifests/debian/deb_folder/rules | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/puppet-manifests/debian/deb_folder/rules b/puppet-manifests/debian/deb_folder/rules
index ec2bca6c..f5ba9a01 100755
--- a/puppet-manifests/debian/deb_folder/rules
+++ b/puppet-manifests/debian/deb_folder/rules
@@ -1,6 +1,8 @@
 #!/usr/bin/make -f
 #export DH_VERBOSE = 1
 
+export DEB_HOST_ARCH = $(shell dpkg-architecture -qDEB_HOST_ARCH 2>/dev/null)
+
 %:
 	dh $@
 
@@ -10,6 +12,9 @@ override_dh_usrlocal:
 	echo "Do Nothing"
 
 override_dh_install:
+ifeq ($(DEB_HOST_ARCH),arm64)
+	sed -i "s/amd64/arm64/" $(CURDIR)/bin/puppet-update-grub-env.py
+endif
 	$(MAKE) install hiera_v5=true ignore_puppet_warnings=true \
 		BINDIR=$(CURDIR)/debian/tmp/usr/local/bin \
 		CONFIGDIR=$(CURDIR)/debian/tmp/etc/puppet \
-- 
2.30.2

